simplex.py

```diff
@@ -170,6 +170,8 @@ def run(self, r_data=None):
                 ff.score = compare.compare_data(r_data, data)
                 opt.pretty_ff_results(ff)
         self.new_ffs = sorted(self.new_ffs + [ff_copy], key=lambda x: x.score)
+        # Allow 3 cycles w/o change for each parameter present
+        self.max_cycles_wo_change = 3*(len(self.new_ffs)-1)
```

Comment:
OK, figure out the naming conventions as you wish, but that particular variable should not be set until you know the number of parameters.

From: Eric Hansen [mailto:notifications@github.com]
Sent: den 28 juni 2016 16:29
To: ericchansen/q2mm <q2mm@noreply.github.com>
Cc: Norrby, Per-Ola <Per-Ola.Norrby@astrazeneca.com>; Author <author@noreply.github.com>
Subject: Re: [ericchansen/q2mm] Mysimplex (#36)


In simplex.py<https://github.com/ericchansen/q2mm/pull/36#discussion_r68767171>:

> @@ -170,6 +170,8 @@ def run(self, r_data=None):

>                  ff.score = compare.compare_data(r_data, data)

>                  opt.pretty_ff_results(ff)

>          self.new_ffs = sorted(self.new_ffs + [ff_copy], key=lambda x: x.score)

> +        # Allow 3 cycles w/o change for each parameter present

> +        self.max_cycles_wo_change = 3*(len(self.new_ffs)-1)

Right, it's non-private, which means that if a user sets it in __init__, he expects that it will actually be used.

If you overwrite it, it's acting like a variable for internal use, thus it should be lead by one underscore.

It is not a private variable, so it also should not have doubling leading underscores.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/ericchansen/q2mm/pull/36/files/0b9c5d0ad0a8d1105e4d26f192324c8bf323a928..398b4e1b8e1242703edfcc5b2a40b20047233942#r68767171>, or mute the thread<https://github.com/notifications/unsubscribe/ANVSNDMyC5UE2turMId_dETY-dXs3ielks5qQS-jgaJpZM4JAC7r>.

________________________________

Confidentiality Notice: This message is private and may contain confidential and proprietary information. If you have received this message in error, please notify us and remove it from your system and note that you must not copy, distribute or take any action in reliance on it. Any unauthorized use or disclosure of the contents of this message is not permitted and may be unlawful.